spikeextractors/extractors/intanrecordingextractor/intanrecordingextractor.py

```diff
@@ -61,16 +63,29 @@ def get_sampling_frequency(self):
         return self._fs
 
     @check_get_traces_args
-    def get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_scaled=True, dtype=None):
+    def get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_scaled: Optional[bool] = None,
```

Comment:
@CodyCBakerPhD can you make rerurn_scaled not optional and set it to True by default?

Reply:
@alejoe91 I played around with that at first but it makes it more complicated to error check without being able to tell if the user manually specified it as True or False, or if it is in default mode. Also note that it is always 'optional' throughout the repo, since it has a default behavior (True). That behavior is mirrored here, and everything in the default mode acts as though it were True. In order to provide the most robust checking against the potentially conflicting `dtype` argument, it really helps to be able to tell when the user sets it, which we can't tell if we just leave it as `True` and allow `dtype` to pass along with an unspecified `return_scale`.

For example, a user that wanted to try to get the base `int` type returned might call

```
my_intan_re = se.IntanRecordingExtractor(intan_file_path)
my_intan_re.get_traces(dtype='int16')
```

but if `return_scaled=True` by default, it would either have to get overridden by the `dtype` (which might imply that the result is scaled, but also somehow an `int` type), or return an error telling the user they have to manually specify both `return_scaled` and `dtype` at `get_traces`. Likewise they could do

```
my_intan_re = se.IntanRecordingExtractor(intan_file_path, dtype='int16')
my_intan_re.get_traces() or my_intan_re.get_traces(return_scaled=True)
```

which, if `return_scaled=True` by default, would force adherence to that and return float traces, which could be quite confusing.

Basically, the way it works right now is, by detecting which arguments were actually specified by the user, we can both provide a clear path of which arguments override each other based on what the user is trying to do (i.e., if `return_scaled` is specified but `dtype` is not, use the implied behavior of `return_scaled`; if `dtype` is specified but `return_scaled` is not, use the `dtype` behavior). If neither is specified in `get_traces` we use the internal `dtype` set at initialization. If both are passed, and conflict with each other, an informative message is given in this respect.

Reply:
Thanks @CodyCBakerPhD, but in this case `return_scaled: Optional[bool] = None` means that if the user doesn't pass it, it is `None`. IMO, in order to get `int16`, the user should do:
```
my_intan_re.get_traces(return_scaled=False, dtype='int16')
```

Since `return_scaled` should be `True` by default, I think that:
```
my_intan_re.get_traces(dtype='int16')
```

Should throw an error. In case `return_scaled` is `False` but `dtype` is not set, I would return `uint16`, which is how the data is stored. 

Also, since we have all this options for `get_traces`, I think we can remove `dtype` as an argument to the `__init__`. What do you think?

Reply:
> Thanks @CodyCBakerPhD, but in this case return_scaled: Optional[bool] = None means that if the user doesn't pass it, it is None.

Yes - but if you read the code below, if it is `None` there are a couple ways it then defaults based on if `dtype` was specified. If you track the default mode from top-down (including `__init__`), it delivers exactly the same as every other extractor as if `return_scaled` was `True`, but we now have that extra piece of information that it was actually passed by the user and not simply defaulted.

While I'm not against removing `dtype` from `__init__` since it really doesn't do much, just keep in mind that removing that breaks back-compatibility, which I'm really trying to preserve here. Also, if we did that, the default `dtype` at `__init__` would have to be `uint16` in order to preserve the gain and offset information.

Another break in back-compatibility also occurs with what you propose on forcing the pairing of `return_scaled=False` with `dtype='int16'` when, under your version, `return_scaled=True` by default. That break I am against, since I know we already make use of that exact function call at several places in currently supported projects, so it would be quite the annoyance to go back in and correct them all. And if it's annoying for me, who knows how many other downstream users would have their workflows disrupted by it.

Reply:
I agree that we should keep in mind backcompatibility, but note that `return_scaled` already breaks some stuff in other extractors (e.g. NWB write function now requires it..). I think that SI is still immature enough to properly make these changes that would make the API better as a whole!